Add `strong_em_symbol` and `newline` options to the converter

andredelft opened the pull request:
Hi,

Thanks for this great package! I like it a lot, though two options I would have liked to see are the following:
1. Both symbols `*` and `_` can be used in the same manner to encode bold and italics in markdown. While they are equivalent, I thought it would be nice to be able to switch between them (this converter only uses `*`). I have added the option `strong_em_symbol` to pass the prefered symbol (defaults to `*`)
2. While `   \n` (two spaces and a newline) is the standard for encoding a line break in markdown, it not always preferred, both because it is not very visible and because many editors trim whitespaces at the end of a line. A popular alternative is to use `\\n` (a backslash and a newline), which also many interpreters support. I have added the option `newline` to the converter, which can switch to this backslash convention by setting it to `'backslash'`.

This PR achieves these things, and they have been documented in the README.

AlexVonB commented:
Hi @andredelft 
thank you very much for your PR, thats a great idea! Would you mind adding some tests, covering the new functionality? I'll merge after this is done :)
Best, Alex

andredelft commented:
Hi @AlexVonB,

It has been a while, sorry for that, but I have added the tests you requested. I also changed the `newline` option to `newline_style` and handled its options in a similar fashion as `heading_style`.

I also thought it would be a good idea to use a `.lower()` transform on both `heading_style` and `newline_style`, so that both options would also correctly interpret the uppercase strings `'ATX'` and `'BACKSLASH'` respectively, as opposed to just their lowercase equivalents. Cf. the commit 7ee87b1.

AlexVonB commented:
Hi @andredelft 

looks great, thank you! It would be nice to have the star/underscore parameter as CONSTANT, just like heading_style and newline_style. I understand if you don't want to invest more time, that's something i can do after the merge. I just wanted your opinion on that. Let me know!

Best, Alex

andredelft commented:
Right, great suggestion! I have pushed a commit to this branch that does this. Feel free to edit it to your likes after the merge!

Best,

André

AlexVonB commented:
Love it, thank you very much!
